### PR TITLE
migrate to jhipster v8

### PR DESCRIPTION
**Pull request by mshima**
Fixes https://github.com/jhipster/generator-jhipster-quarkus/issues/294

### CODE REVIEW

**Comment by mshima** [comment]
The remaining failure is related to keycloak 19 -> 22:
```
2023-11-01 03:42:56,873 INFO  [tc.qua.io/.0.5] (docker-java-stream--304219373) STDOUT: 2023-11-01 03:42:56,871 WARN  [org.keycloak.events] (executor-thread-1) type=LOGIN_ERROR, realmId=jhipster, clientId=web_app, userId=null, ipAddress=172.17.0.1, error=not_allowed, auth_method=oauth_credentials, grant_type=password, client_auth_method=client-secret
2023-11-01 03:42:57,066 INFO  [http-problem] (executor-thread-1) status=401, title="Unauthorized"
Error:  Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 88.54 s <<< FAILURE! -- in io.github.jhipster.sample.web.rest.AccountResourceIT
Error:  io.github.jhipster.sample.web.rest.AccountResourceIT.should_allow_authenticated_user_to_retrieve_own_account_details -- Time elapsed: 2.008 s <<< FAILURE!
java.lang.AssertionError: 
1 expectation failed.
Expected status code <200> but was <400>.
```

https://github.com/jhipster/generator-jhipster-quarkus/blob/2abf8d83b979f58352ead1096d973c9031c94a66/generators/server/templates/quarkus/src/test/java/package/web/rest/AccountResourceIT_oauth2.java.ejs#L33-L46

Maybe related to legacy `grant_type=password`?
The generated backend code is identical (almost, v7 -> v8 changes like package-info auto generation, some relationship changes, etc).

@mraible not sure what's the fix.

**Comment by mraible** [comment]
@mshima Do you happen to have steps to reproduce? If we're using `grant_type=password` in code, we should fix that. It bypasses the whole point of OAuth, IMHO.

**Comment by mraible** [comment]
@vishal423 Do you have any suggestions?

**Comment by mshima** [comment]
@mraible that test seems to login at keycloak and simulates getting the account details from '/account' endpoint with that authentication token.
This test probably would require a reimplementation to drop `grant_type=password` and simulate the entire authentication workflow.
Or `grant_type=password` should be enabled at the keycloak test container.
IMO the test should be dropped and `/account` should be tested by others means.

I can revert keycloak testcontainer to v19 and use copy the keycloak realm from v7.9.4 to use at the test.
What do you think?

**Comment by mraible** [comment]
> I can revert keycloak testcontainer to v19 and use copy the keycloak realm from v7.9.4 to use at the test.

I think this is a good way to test if the Keycloak configuration has changed. 

> IMO the test should be dropped and /account should be tested by others means.

I agree since it's not testing functionality that's used in a JHipster-generated app.

**Comment by mshima** [comment]
@mraible tests are passing now.
I need to do more adjusts post merged adjust like change the diff logic and update to node 18.

File: .blueprint/generate-sample/generator.mjs [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { readdir } from 'node:fs/promises';
+import BaseGenerator from 'generator-jhipster/generators/base';
+import command from './command.mjs';
+
+export default class extends BaseGenerator {
+    sampleName;
```
Comment (mraible) [inline .blueprint/generate-sample/generator.mjs:6]: Should this be 2-space indents instead of 4?

File: .blueprint/generate-sample/templates/samples/imperative-ngx-jwt-maven-mysql-redis.jdl [hunk 2]
```diff
@@ -4,6 +4,7 @@ application {
     authenticationType jwt
     baseName jhipsterSampleApplication
     buildTool maven
+    creationTimestamp 1617901618891
```
Comment (mraible) [inline .blueprint/generate-sample/templates/samples/imperative-ngx-jwt-maven-mysql-redis.jdl:7]: Why is this necessary? I'm guessing it's not a required field.

File: .github/workflows/smoke-test-imperative.yml [hunk 3]
```diff
@@ -53,20 +57,21 @@ jobs:
         uses: actions/checkout@v3
         with:
           repository: 'jhipster/generator-jhipster'
-          ref: v7.9.3
+          ref: v8.0.0-rc.1
```
Comment (mraible) [inline .github/workflows/smoke-test-imperative.yml:60]: Should this be 8.0.0?

File: .prettierrc [hunk 4]
```diff
@@ -1,13 +1,15 @@
...
-  "arrowParens": "avoid",
-  "overrides": [{
-    "files": ".github/**/*.yml",
-    "options": {
-      "tabWidth": 2
-    }
-  }]
+    "printWidth": 140,
+    "singleQuote": true,
+    "tabWidth": 4,
```
Comment (mraible) [inline .prettierrc:4]: Does this set the spaces to 4 instead of 2?

File: .yo-rc.json [hunk 5]
```diff
@@ -1,10 +1,126 @@
 {
+    "generator-jhipster": {
```
Comment (mraible) [inline .yo-rc.json:2]: Should this be indented 2 spaces instead of 4?

File: generators/constants.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { readFileSync } from 'fs';
+import { join } from 'path';
+import { fileURLToPath } from 'url';
+
+export const DEFAULT_DATA_ACCESS = 'activeRecord';
+export const QUARKUS_VERSION = '3.3.0';
```
Comment (mraible) [inline generators/constants.js:6]:
```suggestion
export const QUARKUS_VERSION = '3.5.0';
```
Or this PR might fix it: https://github.com/jhipster/generator-jhipster-quarkus/pull/299

**Review by mraible — APPROVED** [review]

File: .prettierrc (hunk 4 above)
Comment (mshima) [reply .prettierrc:4]: Yes, this is the blueprint old configuration.
I didn't change.
https://github.com/jhipster/generator-jhipster-quarkus/blob/2abf8d83b979f58352ead1096d973c9031c94a66/.prettierrc#L4

Comment (mshima) [reply .prettierrc:4]: It's up to the maintainer to decide indentation.

File: generators/constants.js (hunk 6 above)
Comment (mshima) [reply generators/constants.js:6]: We should avoid mixing jhipster v8 with features like quarkus update.

File: .yo-rc.json (hunk 5 above)
Comment (mshima) [reply .yo-rc.json:2]: prettier config.

File: .blueprint/generate-sample/templates/samples/imperative-ngx-jwt-maven-mysql-redis.jdl (hunk 2 above)
Comment (mshima) [reply .blueprint/generate-sample/templates/samples/imperative-ngx-jwt-maven-mysql-redis.jdl:7]: makes the diff in CI compare step smaller.

File: .blueprint/generate-sample/generator.mjs (hunk 1 above)
Comment (mshima) [reply .blueprint/generate-sample/generator.mjs:6]: prettier config

File: .github/workflows/smoke-test-imperative.yml (hunk 3 above)
Comment (mshima) [reply .github/workflows/smoke-test-imperative.yml:60]: I will try to drop in the following PR.

File: .prettierrc (hunk 4 above)
Comment (mshima) [reply .prettierrc:4]: It's quite easy to change. Change this file and run `npx prettier . --write`.

File: generators/constants.js (hunk 6 above)
Comment (mraible) [reply generators/constants.js:6]: I agree.
